aten/src/ATen/templates/TensorBody.h

```diff
@@ -227,6 +227,10 @@ class CAFFE2_API Tensor {
   // it reports the memory the tensor would take *if* it were contiguous.
   // Defined to be numel() * itemsize()
   size_t nbytes() const {
+    AT_CHECK(layout () != at::kSparse,
```

Comment:
is this used anywhere?  Make it sure it's fast if it's used in someplace important.

Reply:
I checked and it's used in two places:
1) In the quantized constructor, because they don't use normal copy for some reason (use memcpy instead).
2) In the java bindings to create a byte buffer.

Since neither of these need fast nbytes, we are going to commit this.